streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -78,14 +78,24 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
+    private final String storeName;
 
     private boolean sendOldValues = false;
 
+
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
                       Set<String> sourceNodes) {
-        this(topology, name, processorSupplier, sourceNodes, null, null);
+        this(topology, name, processorSupplier, sourceNodes, null, null, null);
+    }
+
+    public KTableImpl(KStreamBuilder topology,
```

Comment:
@guozhangwang the latest PR materializes the source topic too, please have a look if you can.